DallasTemperature.cpp

```diff
@@ -339,19 +339,15 @@ void DallasTemperature::requestTemperatures()
 // returns TRUE  otherwise
 bool DallasTemperature::requestTemperaturesByAddress(uint8_t* deviceAddress)
 {
+  uint8_t bitResolution = getResolution(deviceAddress);
+  if (bitResolution == 0) return false; //Device disconnected
 
   _wire->reset();

```

Comment:
if the root cause is in the Wire lib one can catch the return value of reset too

    int b = _wire->reset();
    if (b == 0) return false;